src/plugins/codecZip.nim

```diff
@@ -238,11 +238,18 @@ proc zipHandleWrite*(self: Plugin, chalk: ChalkObj, encoded: Option[string])
                    {.cdecl.} =
   let injectBinary = attrGet[bool]("zip.inject_binary")
   if injectBinary:
-    try:
-      info(chalk.name & ": Inserting binary into zip archive")
-      insertChalkBinaryIntoZip(chalk)
-    except:
-      error(chalk.name & ": failed to insert chalk binary due to: " & getCurrentExceptionMsg())
+    let
+      ext = chalk.fsRef.splitFile().ext.strip()
+      allowedExtensions = attrGet[seq[string]]("zip.allowed_extensions")
+
+    if ext.startsWith(".") and ext[1..^1] in allowedExtensions:
```

Comment:
this wont work for multi-segment extensions. for example:

```nim
nim> "/foo/bar/baz.tar.gz".splitFile().ext
.gz == type string
```

in other places in chalk we handle that by checking if the filename ends with any of the extensions in the config file:

https://github.com/crashappsec/chalk/blob/8ec46f7abaab712c5c47632d4682d3b03edfac1f/src/plugins/codecCerts.nim#L142-L146

not sure if this is applicable to the zip codec but might as well make it flexible to handle those extensions just in case?

Reply:
refactored to follow the referenced pattern